src/nvtxw.rs

```diff
@@ -0,0 +1,427 @@
+use std::collections::BTreeMap;
+use std::ffi::CString;
+use std::ffi::{c_char, c_void};
+use std::io;
+use std::iter::zip;
+use std::mem::size_of;
+use std::path::PathBuf;
+use std::ptr::{null, null_mut};
+
+use nvtxw::nvtxw;
+
+use crate::data::{DataSourceInfo, EntryID, EntryIndex, EntryInfo, SlotMetaTile, SlotTile, TileID};
+use crate::deferred_data::{CountingDeferredDataSource, DeferredDataSource};
+
+const LEGION_DOMAIN_NAME: &str = "Legion";
+
+pub struct NVTXW<T: DeferredDataSource> {
+    data_source: CountingDeferredDataSource<T>,
+    output: Option<PathBuf>,
+    merge: Option<PathBuf>,
+    zero_time: i64,
+}
+
+type ResultVec = Vec<(EntryID, String, String)>;
+type UnmatchedTileHold = BTreeMap<EntryID, SlotTile>;
+type UnmatchedMetaTileHold = BTreeMap<EntryID, SlotMetaTile>;
```

Comment:
Instead of two maps I think you could make this a `BTreeMap<EntryID, (Option<SlotTile>, Option<SlotMetaTile>)>`. At that point you can do a single pass with [`retain`](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.retain) and trigger only on the entries that contain `(Some(_), Some(_))` (i.e., both sides are filled).

Reply:
I've made this change to the code. How does this part look now?

Reply:
I added a comment on one place that can be simplified, but otherwise it looks good. Thanks.

Reply:
Great!